app/graphql/local-schema.gql

```diff
@@ -15,7 +15,13 @@ extend type Query {
   hiddenBalanceToolTip: Boolean!
   price: String # FIXME test only?
   beta: Boolean!
-  darkMode: Boolean!
+  darkMode: AppearanceMode!
+}
+
+enum AppearanceMode {
```

Comment:
I've brought this up before, but even if we switch to not clearing the apollo cache on every update, I'm still not sure if this is where the data should live. It seems like it would be jarring if a user logs out and then all of a sudden the theme switches.

Reply:
> It seems like it would be jarring if a user logs out and then all of a sudden the theme switches.

I don't think it's really a big issue, for a couple of reasons:
- users don't log out very frequently. devs do it during the development process, but user typically never ever log out
- users typically don't change the dark mode settings from an app. I don't recall ever doing it, expect when I start implementing this functionality, and I saw that WOS have this functionality for instance, but I never played with it before. I think most of the time app rely (exclusively?) on the OS settings for this. As we implement now the OS setting by default, most users would anyway rely on it and probably have little reasons to override the settings

Now, the frequently at which the combination of those elements may occur may be very rare, how critical is the issue? ok an user may see the theme changes on logout, and may be surprise by it, but is it inconvenient or critical in any way? I don't think so.

so all in all, this event is very unlikely with a low effect on user, so I don't think this is worth the complexity of storing the override value outside of where we store all of our other content (graphql).

now maybe it may be worth why I've spent time refactoring the PersistentStateProvider relied on "@react-native-async-storage/async-storage". this means this is not encrypted, which is not ideal for storing password. 

Previously we were using react-native-keychain (created a ticket to be reminded to re-introduce it https://github.com/GaloyMoney/galoy-mobile/issues/848), and this is the right place to store password/token (I don't know why we got rid of it? Maybe you know?). it will be encrypted and can't be extracted from someone who would have physical access to the phone because it uses the secure element in the phone. 

keychain is limiting on how much data can be stored. but currently, as we are back to only have server+password, we could be using back the keychain now. With the previous element where we had a lot more information stored in the json, this was not possible.

this is another reason (and the main one imo) to differentiate where we store token value and the rest of the values in the app. we could have 3 storages, token via keychain, json that is not tied to graphql, and the graphql store, but really I see of complexity doing that and not much downside to not have the "json that is not tied to graphql".

Reply:
Initially I thought the 3 separate storages made conceptual sense, but in reality its probably an indication that the "json not tied to gql" should actually be stored as account settings in the backend. Just as we have language and currency in the backend, it makes sense to store theme there as well. However, I agree that the rarity of the situation may lead to it not being worth the implementation.